src/main/java/com/auction/retailService/controller/StoreResource.java

```diff
@@ -0,0 +1,210 @@
+            @ApiResponse(code = 500, message = "Internal server error")
+    })
+	@GetMapping("/stores")
+	public List<Store> getStores(){
+		return storeService.getStores();
+	}
+	
+	@ApiOperation(value = "Get Store by storeId")
+	@ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Store is retrieved Successfully by storeId"),
+            @ApiResponse(code = 400, message = "Input JSON is wrong"),
+            @ApiResponse(code =401 , message = "User is Not authorized for this operation"),
+            @ApiResponse(code = 412, message = "Validation failed Store Data."),
+            @ApiResponse(code = 500, message = "Internal server error")
+    })
+	@GetMapping("/stores/id/{storeId}")
+	public Optional<Store> getStore(@PathVariable ("storeId") Long storeId) {
+		return storeService.getStoreById(storeId);
+	}
+	
+	@ApiOperation(value = "Get Store by storeName")
+	@ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Store is retrieved Successfully by storeName"),
+            @ApiResponse(code = 400, message = "Input JSON is wrong"),
+            @ApiResponse(code =401 , message = "User is Not authorized for this operation"),
+            @ApiResponse(code = 412, message = "Validation failed Store Data."),
+            @ApiResponse(code = 500, message = "Internal server error")
+    })
+	@GetMapping("/stores/name/{storeName}")
+	public Optional<Store> getStoreByName(@PathVariable ("storeName") @NotNull String storeName) {
+		return storeService.getStoreByName(storeName);
+	}
+	
+	@ApiOperation(value = "Delete Store by storeId")
+	@ApiResponses(value = {
+            @ApiResponse(code =401 , message = "User is Not authorized for this operation"),
+            @ApiResponse(code = 412, message = "Validation failed Store Data."),
+            @ApiResponse(code = 500, message = "Internal server error")
+    })
+	@DeleteMapping("/stores/id/{storeId}")
+	public ResponseEntity<Void> deleteStoreById(@RequestHeader (value="userId" , required=true) Long userId ,@PathVariable ("storeId") Long storeId) {
+		if(userId!=null && userService.isUserExist(userId)) {		
+			if(userService.getUserRole(userId) == RoleConstant.RETAIL_ADMIN || userService.isStoreAdmin(userId,storeId)) {				
+				storeService.deleteStoreById(storeId);
+				return ResponseEntity.noContent().build();
+			}else
+			{
+				throw new UserUnAuthorizedException(ErrorMessageConstant.USER_UNAUTHORIZED_EXCEPTION);
+			}
+		}else
+		{
+			throw new UserNotExistToPerformOperationException(ErrorMessageConstant.USER_NOT_EXIST_FOR_OPERATION);
+		}
+
+	}
+	
+	@DeleteMapping("/stores/name/{storeName}")
+	public ResponseEntity<Void> deleteStoreByName(@RequestHeader (value="userId" , required=true) Long userId ,@PathVariable ("storeName") String storeName) {
+		if(userId!=null && userService.isUserExist(userId)) {		
+			//if(userService.getUserRole(userId) == RoleConstant.RETAIL_ADMIN || userService.isStoreAdmin(userId,storeId)) {				
```

Comment:
Remove commented code like this or leave a TODO explaining the reason why it is there